pkg/querier/queryrange/codec.go

```diff
@@ -828,16 +828,22 @@ func (Codec) MergeResponse(responses ...queryrangebase.Response) (queryrangebase
 		lokiSeriesRes := responses[0].(*LokiSeriesResponse)
 
 		var lokiSeriesData []logproto.SeriesIdentifier
-		uniqueSeries := make(map[string]struct{})
+		uniqueSeries := make(map[uint64]struct{})
 
 		// only unique series should be merged
 		for _, res := range responses {
 			lokiResult := res.(*LokiSeriesResponse)
 			mergedStats.MergeSplit(lokiResult.Statistics)
 			for _, series := range lokiResult.Data {
-				if _, ok := uniqueSeries[series.String()]; !ok {
+				var key uint64
+				if series.Hash != nil {
+					key = series.Hash.Value
+				} else {
+					key = 0 // TODO(karsten): hash labels.
```

Comment:
Ok, just when I created this PR I thought we could add the hashing here. Since `series.String()` is just allocating too much we migh as well feed the labels into a hashing function here.